plugins/main/bridge/bridge.go

```diff
@@ -260,12 +284,20 @@ func cmdAdd(args *skel.CmdArgs) error {
 
 	result.Interfaces = []*current.Interface{brInterface, hostInterface, containerInterface}
 
+	var firstV4Addr net.IP
+	var firstV6Addr net.IP
 	for _, ipc := range result.IPs {
 		// All IPs currently refer to the container interface
 		ipc.Interface = 2
 		if ipc.Gateway == nil && n.IsGW {
 			ipc.Gateway = calcGatewayIP(&ipc.Address)
 		}
+		switch {
+		case firstV4Addr == nil && ipc.Address.IP.To4() != nil:
+			firstV4Addr = ipc.Address.IP
+		case firstV6Addr == nil && len(ipc.Address.IP) == net.IPv6len:
```

Comment:
Ah, yes, good catch. I prefer just adding a To4() check here, rather than cononicalizing. There is a line below (280 old, 312 new) where I think it does make sense to remove the call to To4(), but would appreciate a double-check on the logic.